BitwardenShared/UI/Auth/Extensions/AuthRouter+Redirects.swift

```diff
@@ -354,13 +354,19 @@ extension AuthRouter {
                     return .landingSoftLoggedOut(email: activeAccount.profile.email)
                 }
 
-                // Otherwise, return `.vaultUnlock`.
-                return .vaultUnlock(
-                    activeAccount,
-                    animated: animated,
-                    attemptAutomaticBiometricUnlock: attemptAutomaticBiometricUnlock,
-                    didSwitchAccountAutomatically: didSwitchAccountAutomatically
-                )
+                let biometricUnlockStatus = try await services.biometricsRepository.getBiometricUnlockStatus()
+                let hasMasterPassword = activeAccount.profile.userDecryptionOptions?.hasMasterPassword ?? false
+
+                if case .available(_, true, false) = biometricUnlockStatus, !hasMasterPassword {
+                    return .enterpriseSingleSignOn(email: activeAccount.profile.email)
+                } else {
+                    return .vaultUnlock(
+                        activeAccount,
+                        animated: animated,
+                        attemptAutomaticBiometricUnlock: attemptAutomaticBiometricUnlock,
+                        didSwitchAccountAutomatically: didSwitchAccountAutomatically
+                    )
+                }
```

Comment:
🎨 A small improvement I'd add is that we don't need to check for biometric status if `hasMasterPassword` is `true` so I'd rewrite this as:
```suggestion
                let hasMasterPassword = activeAccount.profile.userDecryptionOptions?.hasMasterPassword == true

                if !hasMasterPassword {
                    let biometricUnlockStatus = try await services.biometricsRepository.getBiometricUnlockStatus()
                    if case .available(_, true, false) = biometricUnlockStatus {
                        return .enterpriseSingleSignOn(email: activeAccount.profile.email)
                    }
                }

                return .vaultUnlock(
                    activeAccount,
                    animated: animated,
                    attemptAutomaticBiometricUnlock: attemptAutomaticBiometricUnlock,
                    didSwitchAccountAutomatically: didSwitchAccountAutomatically
                )
```